File: src/my_py_pkg/my_py_pkg/add_two_ints_client_no_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

def main(args=None):
    rclpy.init(args=args)
    node = Node("add_two_ints_client_no_oop") # name of client

    client = node.create_client(AddTwoInts, "add_two_ints") # type, name of service (same as the server name)
    while not client.wait_for_service(1.0): # stays in loop until server is found. The timeout in the function is the time it waits before looping again
        node.get_logger().warn("Waiting for Add Two Ints server...")
    
    request = AddTwoInts.Request()
    request.a = 9
    request.b = 11

    future = client.call_async(request)             # send request
    rclpy.spin_until_future_complete(node, future)  # wait for response

    response = future.result()                      # set response once recieved
    node.get_logger().info(f"{request.a} + {request.b} = {response.sum}")

    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/add_two_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using namespace std::chrono_literals;

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_no_oop"); //node name
    
    auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints"); // service name
    while(!client->wait_for_service(1s)) // wait until connection is established
    {
        RCLCPP_WARN(node->get_logger(), "Waiting for ther server...");
    }

    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 9;
    request->b = 7;

    auto future = client->async_send_request(request); // send request
    rclcpp::spin_until_future_complete(node, future); // spin until request is sent

    auto response = future.get();
    RCLCPP_INFO(node->get_logger(), "%d + %d = %d", (int)request->a, (int)request->b, (int)response->sum);

    rclcpp::shutdown();
    return 0;
}